### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by blessing-michael**
In this project We;
- Set up React Testing Library.
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components.
- Ensured there are no linter error.

### CODE REVIEW

**Review by zeff96 — COMMENTED** [review]
Hi @blessing-michael , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
It's awesome that you've submitted the project 🚀! However, it seems like we're missing contributions in the codebase from your partner. 😔

Remember, the project requirements call for equal commits from both collaborators. 🤝 Please encourage your partner to dive into the codebase and contribute before our submission. 🙏

Let's make this project shine together! 🌟👩‍💻👨‍💻"

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by topeogunleye** [comment]
@Esmatullah12, please kindly add to the logic also. Adding only to the README file is not a valid pair programming conribution please.

Hi @blessing-michael, please can you get your coding partner to add at least one line of logic.

**Comment by topeogunleye** [comment]
Hi @blessing-michael, @Esmatullah12

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where one partner has only contributed to the README file without making real contributions to the codebase during a pair-programming activity. Our guidelines explicitly state that both partners should collaborate substantively on the codebase. Editing only the README file doesn't qualify as a valid contribution, and as such, your Code Review Request is invalid.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you've made the necessary corrections 🙏.

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/__tests__/utils/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      total: null,
+      next: null,
+      operation: null,
+    });
+  });
+
+  describe('Calculations if buttonName is a Number', () => {
+    // If button is zero and next number is zero, return an empty object
+
+    test('buttonName is zero and next is 0 and No operation', () => {
```
Comment (VICTOR-CHUKWU) [inline src/__tests__/utils/calculate.test.js:31]: - [ ] you did a wonderful job in all your test, good job :clap:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @blessing-michael , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All functions tested correctly
- [ ] all component tested properly
- [ ] clean code base
- [ ] good use of React testing library

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**